app/logic/validator_score_v1_logic.rb

```diff
@@ -209,14 +208,27 @@ def block_history_get
           p.payload[:batch_uuid]
         )
 
-      # TODO: Eliminate the N+1 Query caused by
-      # validator.validator_block_histories.last
+      validator_ids = p.payload[:validators].map { |v| v.id }
+
+      sql = <<-SQL_END
```

Comment:
@brianlong LMK if you can think of a more elegant solution than what I came up with.

Reply:
Try using the :network & :batch_uuid values to grab only the current validator_block_histories. We should have a DB index for [:network, :batch_uuid].